packages/playground/blueprints/src/lib/compile.ts

```diff
@@ -127,9 +131,20 @@ export function compileBlueprint(
 					}
 				}
 
-				for (const { run, step } of compiled) {
-					const result = await run(playground);
-					onStepCompleted(result, step);
+				if (onStepError) {
+					for (const { run, step } of compiled) {
+						try {
+							const result = await run(playground);
+							onStepCompleted(result, step);
+						} catch (error) {
+							onStepError(error, step);
+						}
+					}
+				} else {
+					for (const { run, step } of compiled) {
+						const result = await run(playground);
+						onStepCompleted(result, step);
+					}
```

Comment:
At first I defined a default `onStepError` callback that re-threw the error, but I removed it to maintain current behavior (no try/catch for each step) if the callback is not passed as an option. That way, the error stack trace is cleaner for all other use cases other than the website query API.